test/Index/index_location_directive.swift

```diff
@@ -0,0 +1,11 @@
+// RUN: %empty-directory(%t)
+// RUN: %target-swift-ide-test -print-indexed-symbols -source-filename %s | %FileCheck %s
+
+// The index should probably have the presumed location, but it would need to
+// keep the file in that case as well. So for now, just check that it doesn't
```

Comment:
Would it be sufficient to move the occurrence out of the record it is currently in and into the record based on its presumed location? That seems fairly natural, but maybe I'm missing something.

CC @akyrtzi

Reply:
I think we should consistently use, or not use, presumed loc for both cursor-info and and index data. If we want to use presumed loc for both what would that entail, how feasible is it?

Reply:
Should be pretty easy to add for cursor-info. Index I have much less experience with, but Ben's suggestion seems reasonable to me.

I think it's a good idea to merge this now and work on that later if we think it's important. This at least avoids it having a mixture of both. I've changed the comment a little so it doesn't suggest a solution.

Reply:
SGTM, I agree it's still an improvement over the current situation for indexing to use the in-buffer location.